testing/matrix-sdk-integration-testing/src/tests/sliding_sync/room.rs

```diff
@@ -538,3 +550,299 @@ async fn test_room_notification_count() -> Result<()> {
 
     Ok(())
 }
+
+static DROP_TODEVICE: StdMutex<bool> = StdMutex::new(true);
+fn drop_todevice(response: &mut Bytes) {
+    let Ok(mut json) = serde_json::from_slice::<serde_json::Value>(response) else {
+        return;
+    };
+    let Some(extensions) = json.get_mut("extensions").and_then(|e| e.as_object_mut()) else {
+        return;
+    };
+    let Some(to_device) = extensions.remove("to_device") else {
+        return;
+    };
+    if *DROP_TODEVICE.lock().unwrap() {
+        info!("Dropping to_device: {to_device}");
+        *response = serde_json::to_vec(&json).unwrap().into();
+        return;
+    }
+}
+
+struct CustomResponder {
+    client: reqwest::Client,
+    response_preprocessor: fn(&mut Bytes),
+}
+impl CustomResponder {
+    fn new(response_preprocessor: fn(&mut Bytes)) -> Self {
+        Self { client: reqwest::Client::new(), response_preprocessor }
+    }
+}
+impl wiremock::Respond for CustomResponder {
+    fn respond(&self, request: &wiremock::Request) -> wiremock::ResponseTemplate {
+        let mut req = self.client.request(
```

Comment:
Please sprinkle a few comments here and there explaining what we're doing in this function (and why we're doing it).